pkg/controllers/dynakube/controller.go

```diff
@@ -324,6 +324,15 @@ func (controller *Controller) reconcileComponents(ctx context.Context, dynatrace
 
 	log.Info("start reconciling app injection")
 
+	if istioClient != nil {
+		istioReconciler := controller.istioReconcilerBuilder(istioClient)
+
+		err := istioReconciler.ReconcileCSIDriver(ctx, dynakube)
+		if err != nil {
+			return errors.WithMessage(err, "failed to reconcile istio objects for CSI Driver")
+		}
+	}
+
```

Comment:
this is probably not the correct space for it. you need the version reconciliation to happen before you can do this, 

also I think this is only relevant in case of (oneagent) injection is needed, so probably a nicer place to put is the injection reconciler (near where the relevant version reconcilitation is happening)

https://github.com/Dynatrace/dynatrace-operator/blob/b74cbbdb67c0d2784b42c272c6a027c5adcf0d57/pkg/controllers/dynakube/injection/reconciler.go#L70

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/b5566bd5bdf7edc5afefd8d6d43208d1790c1cf8